lib/reservation_tracker.rb

```diff
@@ -0,0 +1,90 @@
+require 'date'
+require 'pry'
+
+class ReservationTracker
+  attr_reader :rooms, :all_reservations, :cost
+
+
+  def initialize
+    @all_reservations = []
+    @rooms = Room.new
+    @all_blocks = []
```

Comment:
The `@rooms` variable is an opportunity to use dependency injection. That would make some of your testing a little easier - you could give it a shorter room list.